mplc/partner.py

```diff
@@ -49,6 +55,38 @@ def corrupt_labels(self, proportion_corrupted):
             new_label[idx_max - 1] = 1.0
             self.y_train[i] = new_label
 
+    def permute_labels(self, proportion_corrupted=1):
+        if not 0 <= proportion_corrupted <= 1:
+            raise ValueError(
+                f"The proportion of labels to corrupted was {proportion_corrupted} but it must be between 0 and 1."
+            )
+        # Select the indices where the label will be off-set
+        n = int(len(self.y_train) * proportion_corrupted)
+        idx = sample(list(range(len(self.y_train))), n)
+        # Generate the permutation matrix to use
+        self.corruption_matrix = np.zeros((self.num_labels, self.num_labels))
+        idx_permutation = np.random.permutation(self.num_labels)
+        self.corruption_matrix[np.arange(self.num_labels), idx_permutation] = 1
+        # Permute the labels
+        self.y_train[idx] = np.dot(self.y_train[idx], self.corruption_matrix.T)
+
+    def random_labels(self, proportion_corrupted=1):
+        if not 0 <= proportion_corrupted <= 1:
+            raise ValueError(
+                f"The proportion of labels to corrupted was {proportion_corrupted} but it must be between 0 and 1."
+            )
+        # Select the indices where the label will be off-set
+        n = int(len(self.y_train) * proportion_corrupted)
+        idx = sample(list(range(len(self.y_train))), n)
+        # Generate the random matrix to use
+        alpha = np.random.random(self.num_labels)
```

Comment:
It is better to set the parameters to ones in order to have an uniform distribution
```suggestion
        alpha = np.ones(self.num_labels)
```